master/internal/checkpoint_gc.go

```diff
@@ -40,6 +40,7 @@ func runCheckpointGCTask(
 	toDeleteCheckpoints []uuid.UUID,
 	checkpointGlobs []string,
 	deleteTensorboards bool,
+	patchCheckpoints bool,
```

Comment:
i could appreciate functional args for some of these bools since it is so easy to pass them wrong as is.

also, patchCheckpoints is a pretty obtuse name the first time you hit this code; you kinda need to read this whole patch to understand what it means. want to add a comment on this function for it? i tried to think of a better name but the best I got was `syncCheckpointResourcesPartialDelete` lol.

Reply:
I noticed earlier we were just checking if glob == fullDeleteGlob. Did we decide that wasn't explicit enough? Or it is the edge case bug where someone passes the fullDeleteGlob exactly that we were concerned about?

Reply:
addressing your points:
- going to add more comments to `runCheckpointGCTask` explaining the function of `patchCheckpoints`
- regarding `glob == fullDeleteGlob`, it got messy/complicated when I was trying to specify a return value in the partial & full delete case that could still be sent over to `patchCheckpoints` where applicable. I think the Python code is clear as it stands, and adding more conditions to `deleteCheckpoints` would obfuscate that